### ISSUE TITLE
Bug: xray_trace_id is not populated in log lines

### ISSUE THREAD

**Issue by humanzz** [aws-powertools/powertools-lambda-typescript#773]
## Bug description

According to [documentation](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/logger/#standard-structured-keys), the `xray_trace_id` should be emitted in all log lines if the request is traced. However, I'm not seeing any setting of that field in my logs despite the `REPORT` log line showing the request is traced.

### Expected Behavior

All log lines to have `xray_trace_id` populated if the request is traced (as confirmed by `REPORT` line).

### Current Behavior

No log lines showing the `xray_trace_id`.

### Possible Solution

I've read a little bit in the code in [Logger.ts](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/logger/src/Logger.ts) and I suspect that [xRayTraceId](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/logger/src/Logger.ts#L632) is set only once at `Logger` construction time.

The traceId is read via an environment variable that'll change for each different request. As such, I think a possible solution is to leverage `addContext` to also set that value when calling [this.addToPowertoolLogData](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/logger/src/Logger.ts#L166)

The actual behaviour I've seen is that `xray_trace_id` is not being written to the logs. If my understanding is correct, I believe this issue can be manifested in 1 of 2 ways
1. The one observed i.e. lack of `xray_trace_id` which is likely to happen if the first request is not traced (i.e. cold start request). In this case all subsequent requests will not have the field
2. Wrong `xray_trace_id` (not observed) if the first request is traced and the environment variables is read and the field is set in Logger's state but subsequent requests will not update that value.

### Steps to Reproduce

For a handler looking like

```
export const handler = middy(vanillaHandler)
  .use(captureLambdaHandler(tracer))
  .use(logMetrics(metrics))
  .use(injectLambdaContext(logger)),
```

1. Invoke the function multiple times
3. Inspect the log lines looking for `xray_trace_id` in the different requests (especially those where `REPORT` line indicates request is traced)

`xray_trace_id` is missing

### Environment

* **Powertools version used**: 0.8.0
* **Packaging format (Layers, npm)**: npm
* **AWS Lambda function runtime:** Node.js 14.x
* **Debugging logs:**

### Related issues, RFCs

N/A

**Comment by dreamorosi** [comment]
Hello @humanzz, thanks for opening this issue.

I made a couple of tests and was able to reproduce the behaviour you describe also in the latest version (`v 0.8.1`) - see screenshot below in which tracing is enabled and both executions were sampled (no `xray_trace_id` present):
![Screenshot 2022-04-15 at 18 03 32](https://user-images.githubusercontent.com/7353869/163593796-18e61ef4-1094-4c98-b3b3-9e129f266d53.png)

We will be looking at a fix for this.

**Comment by dreamorosi** [comment]
After a closer look I just realised that it's not possible to piggy back on `addContext` like suggested by OP. We can guarantee that the `addContext` function is ran only when Logger is used with decorators and middleware, while when running [in manual mode the function has to be called explicitly](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/logger/#capturing-lambda-context-info) by customers. 

The above would contradict with the statement ([docs](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/logger/#standard-structured-keys)) that `xray_trace_id` is part of the standard structured keys.

Will be looking at alternatives.

**Comment by dreamorosi** [comment]
A fix is available in #776, it will be most probably included in one of the next two releases based on time needed for review & merge.

**Comment by humanzz** [comment]
Thanks @dreamorosi, really appreciate the prompt action. I'll keep an eye for the upcoming releases
